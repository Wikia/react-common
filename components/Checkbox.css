.wds-checkbox {
  display: inline-flex;
}
.wds-checkbox__input {
  display: none;
}
.wds-checkbox__label {
  cursor: pointer;
  display: inline-flex;
}
.wds-checkbox__icon, .wds-checkbox__lock-icon {
  margin-right: 13px;
}
.wds-checkbox__input[disabled] + .wds-checkbox__label {
  color: #7f9998;
}