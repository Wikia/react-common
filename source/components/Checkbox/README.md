Defaults:
```js
initialState = {
    checkbox1: true,
    checkbox2: false,
};
<div>
    <Checkbox checked={state.checkbox1} onChange={() => setState({checkbox1: !state.checkbox1})}>Default checkbox</Checkbox>
    <Checkbox className="extra-class-name" checked={state.checkbox2} onChange={() => setState({checkbox2: !state.checkbox2})}>Extra className</Checkbox>
</div>
```

Disabled checkbox:
```js
initialState = {
    checkbox3: false,
    checkbox4: true,
};

<div>
    <Checkbox disabled checked={state.checkbox3} onChange={() => setState({checkbox3: !state.checkbox3})}>Disabled unchecked checkbox</Checkbox>
    <Checkbox disabled checked={state.checkbox4} onChange={() => setState({checkbox4: !state.checkbox4})}>Disabled checked checkbox</Checkbox>
</div>
```

