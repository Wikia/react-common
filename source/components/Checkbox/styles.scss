.wds-checkbox {
	display: inline-flex;

	&__input {
		display: none;
	}

	&__label {
		cursor: pointer;
		display: inline-flex;
	}

	&__icon {
		margin-right: 13px;
	}
}
