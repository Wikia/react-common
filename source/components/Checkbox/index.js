import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import uniqueId from 'lodash/uniqueId';

import IconCheckboxSmall from '../../icons/IconCheckboxSmall';
import IconCheckboxEmptySmall from '../../icons/IconCheckboxEmptySmall';
import IconLockSmall from '../../icons/IconLockSmall';

import './styles.scss';

class Checkbox extends React.Component {
    constructor(props) {
        super(props);

        this.id = uniqueId('wds-checkbox-');
    }

    render() {
        const {
            children,
            className,
            checked,
            disabled,
            onChange,
        } = this.props;

        return (
            <div className={classNames('wds-checkbox', className)}>
                <input
                    className="wds-checkbox__input"
                    type="checkbox"
                    id={this.id}
                    checked={checked}
                    onChange={onChange}
                    disabled={disabled}
                />
                <label className="wds-checkbox__label" htmlFor={this.id}>
                    {checked && !disabled && <IconCheckboxSmall className="wds-icon wds-icon-small wds-checkbox__icon" />}
                    {!checked && !disabled && <IconCheckboxEmptySmall className="wds-icon wds-icon-small wds-checkbox__icon" />}
                    {disabled && <IconLockSmall className="wds-icon wds-icon-small wds-checkbox__lock-icon" />}
                    {children}
                </label>
            </div>
        );
    }
}

Checkbox.propTypes = {
    /** Is checkbox checked */
    checked: PropTypes.bool,
    /** Label passed as a child */
    children: PropTypes.node,
    /** Additional class name */
    className: PropTypes.string,
    /** Is checkbox disabled */
    disabled: PropTypes.bool,
    /** Function triggered when checkbox changes its state */
    onChange: PropTypes.func.isRequired,
};

Checkbox.defaultProps = {
    checked: false,
    children: null,
    className: undefined,
    disabled: false,
};

export default Checkbox;
