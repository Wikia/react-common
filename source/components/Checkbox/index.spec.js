import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import Checkbox from './index';

/* eslint-disable no-alert */

test('Checkbox renders correctly with default values', () => {
    const component = renderer.create(
        <Checkbox onChange={() => {}} />,
    );
    expect(component.toJSON()).toMatchSnapshot();
});

test('Checkbox renders correctly checked (and with label and class name)', () => {
    const component = renderer.create(
        <Checkbox checked onChange={() => {}} className="checked-box">a Checkbox</Checkbox>,
    );
    expect(component.toJSON()).toMatchSnapshot();
});

test('Checkbox renders correctly when disabled', () => {
    const component = renderer.create(
        <Checkbox disabled onChange={() => {}}>disabled checkbox</Checkbox>,
    );
    expect(component.toJSON()).toMatchSnapshot();
});


test('Checkbox is not clickable when disabled', () => {
    const mockonChange = sinon.spy();
    const wrapper = shallow(
        <Checkbox disabled onChange={mockonChange}>disabled</Checkbox>
    );

    wrapper.find('input').simulate('click');

    expect(mockonChange.notCalled).toBe(true);
});

test('Checkbox triggers onChange event', () => {
    const mockonChange = sinon.spy();
    const wrapper = shallow(
        <Checkbox onChange={mockonChange}>clickable</Checkbox>
    );

    wrapper.find('input').simulate('change');

    expect(mockonChange.calledOnce).toBe(true);
});
