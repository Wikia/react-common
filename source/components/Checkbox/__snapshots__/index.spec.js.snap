// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly checked (and with label and class name) 1`] = `
<div
  className="wds-checkbox checked-box"
>
  <input
    checked={true}
    className="wds-checkbox__input"
    disabled={false}
    id="wds-checkbox-2"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="wds-checkbox__label"
    htmlFor="wds-checkbox-2"
  >
    <div
      className="wds-icon wds-icon-small wds-checkbox__icon"
    />
    a Checkbox
  </label>
</div>
`;

exports[`Checkbox renders correctly when disabled 1`] = `
<div
  className="wds-checkbox"
>
  <input
    checked={false}
    className="wds-checkbox__input"
    disabled={true}
    id="wds-checkbox-3"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="wds-checkbox__label"
    htmlFor="wds-checkbox-3"
  >
    <div
      className="wds-icon wds-icon-small wds-checkbox__lock-icon"
    />
    disabled checkbox
  </label>
</div>
`;

exports[`Checkbox renders correctly with default values 1`] = `
<div
  className="wds-checkbox"
>
  <input
    checked={false}
    className="wds-checkbox__input"
    disabled={false}
    id="wds-checkbox-1"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="wds-checkbox__label"
    htmlFor="wds-checkbox-1"
  >
    <div
      className="wds-icon wds-icon-small wds-checkbox__icon"
    />
  </label>
</div>
`;
